Modules/posixmodule.c

```diff
@@ -5219,19 +5220,19 @@ os_posix_spawn_impl(PyObject *module, path_t *path, PyObject *argv,
                         goto exit;
                     }
 
-                    long open_fd = PyLong_AsLong(PySequence_GetItem(file_actions_obj, 1));
+                    long open_fd = PyLong_AsLong(PySequence_Fast_GET_ITEM(file_actions_obj, 1));
```

Comment:
`PyLong_AsLong()` can call arbitrary code. If `file_actions` is a list, the size of `file_actions_obj` can be changed after this, and the following `PySequence_Fast_GET_ITEM()` could read past the end of the list.

I would require `file_actions` to be a tuple and use `PyArg_ParseTuple()` for parsing it.